dspace-server-webapp/src/main/java/org/dspace/app/rest/security/jwt/JWTTokenRestAuthenticationServiceImpl.java

```diff
@@ -80,9 +86,13 @@ public void addAuthenticationDataForUser(HttpServletRequest request, HttpServlet
 
             String token = loginJWTTokenHandler.createTokenForEPerson(context, request,
                                                                  authentication.getPreviousLoginDate(), groups);
+            context.commit();
 
+            // Add newly generated auth token to the response
             addTokenToResponse(response, token, addCookie);
-            context.commit();
+
+            // Reset our CSRF token, generating a new one
+            resetCSRFToken(request, response);
```

Comment:
@tdonohue it is not clear to me what is done here, can you help?
Are we generating a token on line 87 and immediately generate a new one in line 95? if this is the case, is not the token in the cookie different from the one send via header in line 95?

Reply:
@abollini : You've confused the two tokens here.  Line 92 (`addTokenToResponse`) is adding the *Auth* Token (JWT) to the response.  Line 95 (`resetCSRFToken`) is resetting the *CSRF* Token.  So, these are two entirely separate tokens.  The logic here is that anytime you create a *new* auth token (JWT) we should immediately reset the CSRF Token....this ensures that the CSRF token changes anytime you login.